single_epoch.py

```diff
@@ -7,14 +7,22 @@
 from torch.nn import Module
 from torch.optim import Optimizer
 from torch_geometric.data import HeteroData, Data
-from tqdm import tqdm
+from tqdm.autonotebook import tqdm
 from sklearn.metrics import roc_auc_score
 from torch_geometric.loader import NeighborLoader, LinkNeighborLoader
 from utils.metrics_encoder_decoder import get_metrics_universal
 from utils.get_info import select_properties
+from utils.python_type import isnotebook
 
 device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
 
+# if isnotebook():
```

Comment:
is this then not necessary?

Reply:
Yes, you are right. We do either or. To be honest the original version gave more flexibility as we were checking for systems manually.